### PR TITLE
collect generic helper/utility functions in utils.R

### PR DESCRIPTION
**Pull request by kkmann**
# Description

suggesting to consolidate output.R and parallel.R in utils.R. I guess it makes sense to keep the helper function from satterthwaite.R in the same file since they are specific.

### CODE REVIEW

**Comment by kkmann** [comment]
see #98

**Review by danielinteractive — APPROVED** [review]

**Comment by kkmann** [comment]
looks like the gh actions run into issues with git checkout. not sure what's causing this. R CMD check runs perfectly fine locally.

**Comment by danielinteractive** [comment]
@cicdguy any idea what is going on re: checks?

**Comment by cicdguy** [comment]
> @cicdguy any idea what is going on re: checks?
Let me take a look.

**Comment by cicdguy** [comment]
Ok I see the issue. Will need to create a separate PR for it.
The issue is that the GitHub token used for those workflows do not work for forks (for security reasons). 
I'll just replace the tokens with the generic ones.

**Comment by cicdguy** [comment]
> Ok I see the issue. Will need to create a separate PR for it. The issue is that the GitHub token used for those workflows do not work for forks (for security reasons). I'll just replace the tokens with the generic ones.

Actually, never mind. I do see that the token is generic. It's got something to do with token permissions.

File: R/utils.R [hunk 1]
```diff
@@ -1,3 +1,56 @@
...
+  list(
+    result = result,
+    warnings = setdiff(env$warning, remove$warnings),
+    errors = setdiff(env$error, remove$errors),
+    messages = setdiff(env$message, remove$messages)
+  )
+}
+
+
+
```
Comment (cicdguy) [inline R/utils.R:53]:
```suggestion
```

**Comment by cicdguy** [comment]
FYI @kkmann - feel free to revert/drop my commit. I ended up fixing the issue and the workflows should run fine now.

File: R/utils.R [hunk 2]
```diff
@@ -1,3 +1,55 @@
...
+  )
+  list(
+    result = result,
+    warnings = setdiff(env$warning, remove$warnings),
+    errors = setdiff(env$error, remove$errors),
+    messages = setdiff(env$message, remove$messages)
+  )
+}
+
+
```
Comment (cicdguy) [inline R/utils.R:52]:
```suggestion

```

**Comment by danielinteractive** [comment]
Ah good point. @kkmann actually no need to fork - you can push feature branches etc on this repo. If that does not work please let us know

**Comment by kkmann** [comment]
thanks, guess these are still toothing issues from seeting this free from you Roche internal dev ops.

**Pull request merged**